packages/crud/src/vaadin-crud.js

```diff
@@ -590,6 +570,42 @@ class Crud extends ElementMixin(ThemableMixin(PolymerElement)) {
     }
   }
 
+  /** @protected */
+  get slots() {
+    // NOTE: order in which the toolbar buttons are listed matters.
+    return {
+      ...super.slots,
+      header: () => {
+        return document.createElement('h3');
+      },
+      form: () => {
+        const form = document.createElement('vaadin-crud-form');
+        if (this.theme) {
+          form.setAttribute('theme', this.theme);
+        }
```

Comment:
This shouldn't be needed if `__formChanged` takes care of propagating the theme. Would be safer to add `_form` to the list of dependencies in `__formChanged(theme, include, exclude, _form)` to make sure the function invokes on form change.

Similar change could be done for `__setEditorHeader` so you wouldn't need to invoke the observer function manually.

Reply:
Good point, done.